src/history.js

```diff
@@ -4,9 +4,10 @@ const HISTORY_KEY = "save-in-history";
 
 const SaveHistory = {
   add: async (entry) => {
-    const current = (await browser.storage.local.get(HISTORY_KEY)) || {};
-    await browser.storage.local.set({
-      [HISTORY_KEY]: [...(current.history || []), entry],
+    browser.storage.local.get(HISTORY_KEY, function(current) {
```

Comment:
I'll need to try this out, `local.get` is async and returns a promise as per https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea/get

MDN also doesn't seem to see to indicate a callback function as the second argument.

Reply:
@gyng If you look on bottom of that page, it's actually mentioned there as example: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea/get#chrome_examples (with different function syntax, but still).

Or I can rewrite it to promise using `then`. I'm just suspicious about the await. And there is no other await in your extension ...

Reply:
Ok. Another pull request with `then` pushed.